### PR TITLE
Get list of facebook users for chat

### LINKED ISSUE #125: Return null
slap returns null when used in group chats (possibly because of sender name)
returns msg as "!slap <person> fucked up. Damn you null."

### LINKED ISSUE #131: event.senderName is no longer passed by facebook

### PR DESCRIPTION
**Pull request by mrkno**
Add method to determine names of facebook users since new Facebook API update

resolves #131, resolves #125

### CODE REVIEW

**Comment by mrkno** [comment]
Does not resolve the issue described [here](https://github.com/Schmavery/facebook-chat-api/issues/260), however does have a fallback to deal with this case if need be. Not sure if that problem is with us or the underlying API but regardless this is an improvement over our current state in most circumstances.

File: core/output/facebook.js [hunk 1]
```diff
@@ -4,17 +4,61 @@ var fb = require("facebook-chat-api"),
...
 exports.start = function(callback) {
 	fb({email: this.config.username, password: this.config.password}, function (err, api) {
 		if(err) {
 			console.error(err);
 			process.exit(-1);
 		}
-		api.setOptions({listenEvents: true});
+
+		var options = {
+			listenEvents: true
```
Comment (ParasDPain) [inline core/output/facebook.js]: Any reason why `options.logLevel` is not added inline using lambdas as 
`logLevel: console.isDebug() ? 'verbose' : 'silent' `

File: core/output/facebook.js [hunk 2]
```diff
@@ -71,23 +115,54 @@ exports.start = function(callback) {
 					endTyping = null;
 				}
 				api.setTitle(title, thread);
+			},
```
Comment (ParasDPain) [inline core/output/facebook.js]: I also noticed 
```
if (endTyping != null) {
                    endTyping();
                    endTyping = null;
}
```
can be extracted into a separate method as it's being repeated 5 times

File: core/output/facebook.js (hunk 1 above)
Comment (mrkno) [inline core/output/facebook.js]: Was deliberately not done to preserve default behavior. In the documentation for the chat api there isn't a specified default log level, so I was leaving it to the implementors to decide what was appropriate by default.

Technically a tenary not a lambda...

File: core/output/facebook.js (hunk 2 above)
Comment (mrkno) [inline core/output/facebook.js]: 5 times!? How did I not notice this before....

File: core/output/facebook.js (hunk 1 above)
Comment (ParasDPain) [inline core/output/facebook.js]: Oh :+1: good call! And yes a ternary it is

**Comment by mrkno** [comment]
@ParasDPain if you approve of the changes, comment "Approved" and I'll merge it

**Comment by ParasDPain** [comment]
Yep approved!

**Pull request merged**